### PR TITLE
Make some structs private and add some docs

### PR DESCRIPTION
**Pull request by Jarema**
Let's hide what does not have to be public.

### CODE REVIEW

File: async-nats/src/lib.rs [hunk 1]
```diff
@@ -11,6 +11,95 @@
...
+//!     assert_eq!(i, 10);
+//! }
+//!
+//! ```
+//!
+//! ### Publish
+//!
+//! ```
+//! # use bytes::Bytes;
+//! # use futures_util::StreamExt;
```
Comment (caspervonb) [inline async-nats/src/lib.rs:66]: You don't need stream_ext here do you?

**Review by caspervonb — COMMENTED** [review]
One nit

File: async-nats/src/lib.rs (hunk 1 above)
Comment (Jarema) [reply async-nats/src/lib.rs:66]: true :).

**Review by caspervonb — APPROVED** [review]
LGTM (presuming green CI)

**Pull request merged**